includes/Services/FlowService.php

```diff
@@ -0,0 +1,199 @@
+		// check if data is available in the database if not then fetch the default dataset
+		$flow_data              = Flows::get_data();
+		$flow_data['createdAt'] = time();
+		// get current flow type
+		$flow_type = Data::current_flow();
+		// update default data if flow type is ecommerce
+		if($flow_type == 'ecommerce') 
+			$flow_data = self::update_default_data_for_ecommerce( $flow_data );
+		return $flow_data;
+	}
+
+	public static function update_flow_data($params) {
+		$flow_data = FlowService::read_flow_data_from_wp_option();
+		return self::update_post_call_data_recursive($flow_data, $params);
+	}
+
+	/*
+	 * function to update the Flow Data (Blueprint) in an array recursively in comparison to Flows::get_data() (Database)
+	 */	
+	private static function update_flow_data_recursive($default_flow_data, $flow_data) {
+		$flow_data_fixes = Flows::get_fixes();
+		$exception_list = Flows::get_exception_list();
+		$updated_flow_data = array();
+		foreach ($default_flow_data as $key => $value)
+		{ 
+			// To align the datatype of the respective values with the one set in the blueprint
+			if(strcmp(gettype($value), gettype($flow_data[$key])) != 0) {
+				$updated_flow_data[$key] = $value;
+				continue;
+			}
+
+			// Any Key renamed is updated in the database with NewKey and the value from the OldKey is retained or not based on retain_existing_value
+			if($flow_data_fixes) {
+				foreach ($flow_data_fixes as $old_key=>$replace_key) {
+					if (array_key_exists($old_key, $flow_data) && array_key_exists($replace_key['new_key'], $default_flow_data)) {
+						($replace_key['retain_existing_value'])?
+							$flow_data[$replace_key['new_key']] = $flow_data[$old_key]
+							: $flow_data[$replace_key['new_key']] = $default_flow_data[$replace_key['new_key']];
+						unset($flow_data[$old_key]);
+					}	
+				}
+			}
+
+			if(array_key_exists($key, $flow_data)) {
+
+				// Accepts the value of Exception List keys as is from the database OR non array OR empty array values
+				if(($exception_list && isset($exception_list[$key])) || !is_array($value) || (is_array($value) && sizeof($value) === 0))
+					$updated_flow_data[$key] = $flow_data[$key];
+
+				else {
+					// To handle Indexed Arrays gracefully
+					if (count(array_filter(array_keys($value), 'is_string')) === 0) {
+						// To check if an Indexed Array is further Nested or Not
+						foreach($value as $index_key => $index_value) {
+							// For Indexed Arrays having Non Associative Array Values
+							(!is_array($index_value) && isset($flow_data[$key]))? 
+								$updated_flow_data[$key] = $flow_data[$key] 
+								: $updated_flow_data[$key] = $value;
+						}
+						continue;
+					} 
+
+					// To handle Associative Arrays gracefully
+					$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
+				}
+			}
+
+			// Adds or deletes key-value pairs in DB based on the modification in the blueprint
+			else
+				$updated_flow_data[$key] = $value;
```

Comment:
```suggestion
if( ! array_key_exists($key, $flow_data)) {
				$updated_flow_data[$key] = $value;
			}

				// Accepts the value of Exception List keys as is from the database OR non array OR empty array values
			if(isset($exception_list[$key]) || ! is_array($value) || sizeof($value) === 0)
				$updated_flow_data[$key] = $flow_data[$key];
				continue;

			// To handle Indexed Arrays gracefully
			if (count(array_filter(array_keys($value), 'is_string')) === 0) {
				// To check if an Indexed Array is further Nested or Not
				foreach($value as $index_key => $index_value) {
					// For Indexed Arrays having Non Associative Array Values
					(!is_array($index_value) && isset($flow_data[$key]))? 
						$updated_flow_data[$key] = $flow_data[$key] 
						: $updated_flow_data[$key] = $value;
				}
				continue;
			} 

			// To handle Associative Arrays gracefully
			$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
```
Let's always aim to make the code itself readable instead of adding comments that explain what it does